motr/ut/obj.c

```diff
@@ -679,7 +708,7 @@ static void ut_test_m0_entity_create(void)
 			   m0_client_layout_id(instance));
 
 	rc = m0_entity_create(NULL, &obj.ob_entity, &ops[0]);
-	M0_UT_ASSERT(rc = -ENOENT);
```

Comment:
Good catch!

Reply:
Now, I know the reason why the -EINVAL is returned:
the layouts are not initialized. So we can not find layout with the specified lid.

Reply:
So, please add comments to explain why this error is expected.

with the following patch,
```
diff --git a/motr/obj.c b/motr/obj.c
index 9ec61cb2..b4158b48 100644
--- a/motr/obj.c
+++ b/motr/obj.c
@@ -411,7 +411,7 @@ m0__obj_layout_instance_build(struct m0_client *cinst,
         */
        layout = m0_layout_find(&cinst->m0c_reqh.rh_ldom, layout_id);
        if (layout == NULL) {
-               rc = -EINVAL;
+               rc = M0_ERR(-EINVAL);
                goto out;
        }
 
@@ -527,15 +527,17 @@ static int obj_namei_op_init(struct m0_entity *entity,
                                        m0__obj_layout_id_get(oo));
        rc = m0__obj_layout_instance_build(cinst, lid,
                                           &oo->oo_fid, &linst);
-       if (rc != 0)
+       if (rc != 0) {
+               rc = M0_ERR(rc);
                goto error;
+       }
        oo->oo_layout_instance = linst;
 
 #ifdef CLIENT_FOR_M0T1FS
        /* Set the object's parent's fid. */
        if (!m0_fid_is_set(&cinst->m0c_root_fid) ||
            !m0_fid_is_valid(&cinst->m0c_root_fid)) {
-               rc = -EINVAL;
+               rc = M0_ERR(-EINVAL);
                goto error;
        }
        oo->oo_pfid = cinst->m0c_root_fid;
@@ -543,8 +545,10 @@ static int obj_namei_op_init(struct m0_entity *entity,
        /* Generate a valid oo_name. */
        obj_name = m0_alloc(M0_OBJ_NAME_MAX_LEN);
        rc = obj_fid_make_name(obj_name, M0_OBJ_NAME_MAX_LEN, &oo->oo_fid);
-       if (rc != 0)
+       if (rc != 0) {
+               rc = M0_ERR(rc);
                goto error;
+       }
        m0_buf_init(&oo->oo_name, obj_name, strlen(obj_name));
 #endif
        M0_ASSERT(rc == 0);

```
and when you run:
./scripts/m0 run-ut -t obj-ut:"m0_entity_create\(object\)" -e ERROR
You will see the error.

Reply:
Done.